### PR TITLE
User group and roles

### PR DESCRIPTION
**Pull request by itkg-rboessel**

### CODE REVIEW

File: en/user_guide/user.rst [hunk 1]
```diff
@@ -0,0 +1,50 @@
+User, Group and Role
+====================
+
+Open Orchestra offer the possibility to manage different User with different right.
+
+User Management
+---------------
+
+Page Users in Administration allow to search,create,edit ans delete site users.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: Can you put a white space after the comma ?

File: en/user_guide/user.rst [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+At the creation, some fields are required :
+    - First Name
+    - Last Name
+    - E-mail
+    - Username
+    - Password
+
+After it's possible to Edit User to choose this default language which is display on connect and affiliate himself to some group.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: Once the user is created, you can add some information regarding his default language, his group, ...

File: en/user_guide/user.rst [hunk 3]
```diff
@@ -0,0 +1,50 @@
+User, Group and Role
+====================
+
+Open Orchestra offer the possibility to manage different User with different right.
```
Comment (juchi) [inline en/user_guide/user.rst]: offers
users
rights

File: en/user_guide/user.rst (hunk 1 above)
Comment (juchi) [inline en/user_guide/user.rst]: "ans" -> "and"

File: en/user_guide/user.rst [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+---------------
+
+Page Users in Administration allow to search,create,edit ans delete site users.
+
+At the creation, some fields are required :
+    - First Name
+    - Last Name
+    - E-mail
+    - Username
+    - Password
```
Comment (juchi) [inline en/user_guide/user.rst]: bulleted lists are rather made with *

File: en/user_guide/user.rst [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+After it's possible to Edit User to choose this default language which is display on connect and affiliate himself to some group.
+
+Group Management
+----------------
+
+In Open Orchestra, Users are members (or not) of groups.
```
Comment (juchi) [inline en/user_guide/user.rst]: users can be members of groups

File: en/user_guide/user.rst [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+Group give at User some right and access for website administration.
```
Comment (juchi) [inline en/user_guide/user.rst]: A group gives a user some rights

File: en/user_guide/user.rst [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+Each group have a name and is associate to ONE website.
```
Comment (juchi) [inline en/user_guide/user.rst]: has
associated
to only one

File: en/user_guide/user.rst [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+It contain a list of roles for User.
```
Comment (juchi) [inline en/user_guide/user.rst]: contains
for a user ? for users ?

File: en/user_guide/user.rst [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+
+Role Management
+---------------
+
+Role give one type of authorization for User.
```
Comment (juchi) [inline en/user_guide/user.rst]: A role gives
for a user

File: en/user_guide/user.rst [hunk 10]
```diff
@@ -0,0 +1,50 @@
...
+
+Role characteristics
+~~~~~~~~~~~~~~~~~~~~
+
+Roles contains a name used by sensio security bundle to restrict authorization::
```
Comment (juchi) [inline en/user_guide/user.rst]: "A role contains" or "Roles contain"
double column at line end
Actually i would rephrase, in a user doc we should not refer to a bundle or to code

File: en/user_guide/user.rst [hunk 11]
```diff
@@ -0,0 +1,50 @@
...
+    /**
+     * @Security("has_role('ROLE_NAME')")
+     */
+    public function indexAction()
+    {
+        // ...
+    }
+
+
+a description for each site languages to display in admin panel and two optionals parameters :
```
Comment (juchi) [inline en/user_guide/user.rst]: Is this a new sentence ? It's not clear how to read it
optional

File: en/user_guide/user.rst [hunk 12]
```diff
@@ -0,0 +1,50 @@
...
+    - Status from
+    - Status destination
```
Comment (juchi) [inline en/user_guide/user.rst]: - instead of -
  are these actual labels from the BO ? It would rather use "Original status" "Destination status"

File: en/user_guide/user.rst [hunk 13]
```diff
@@ -0,0 +1,50 @@
...
+to sharp roles for specifics content_type and define an edition line.
```
Comment (juchi) [inline en/user_guide/user.rst]: specific content types

**Comment by juchi** [comment]
General remark : entities should not have capital letters in the middle of a sentence.

File: en/user_guide/user.rst [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+* username
+* password
+
+Once the user is created, you can add some information regarding his default language and his group.
+
+Group Management
+----------------
+
+In Open Orchestra, users can be members of groups.
+A group gives a user some rights and access for website administration.
```
Comment (ngilain) [inline en/user_guide/user.rst]: Actually rights can also be given to access some private parts of the front sites, not only on Back Office.

File: en/user_guide/user.rst [hunk 15]
```diff
@@ -0,0 +1,42 @@
+User, Group and Role
+====================
+
+Open Orchestra offers the possibility to manage different users with different rights.
+
+User Management
+---------------
+
+Page users in administration allow to search, create, edit and delete site users.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: You could also add that users are in the same, front or back and they appear in the same list

Comment (juchi) [inline en/user_guide/user.rst]: I would rephrase the sentence also : "The users page allows to search..."

File: en/user_guide/user.rst [hunk 16]
```diff
@@ -0,0 +1,41 @@
+User, Group and Role
+====================
+
+Open Orchestra offers the possibility to manage different users with different rights.
+
+User Management
+---------------
+
+The users page allows to search, create, edit and delete current site users.
```
Comment (ngilain) [inline en/user_guide/user.rst]: I think it's not for "current" site but for all sites

Comment (nicolasThal) [inline en/user_guide/user.rst]: Yes, you can managed users from the whole plateform

File: en/user_guide/user.rst [hunk 17]
```diff
@@ -0,0 +1,41 @@
...
+
+In Open Orchestra, users can be members of groups.
+A group gives a user some rights and access for website.
+Each group has a name and is associated to only one website.
+It contains a list of roles for users.
+
+Role Management
+---------------
+
+A role gives one type of authorization for a user.
```
Comment (ngilain) [inline en/user_guide/user.rst:26]: Maybe you could give several examples of roles plus an example of what a group is. It will help the reader to have a clear knowledge of what you're talking about. Maybe to do that you'll have to switch the order : first definition of a role then definition of a group.

File: en/user_guide/user.rst [hunk 18]
```diff
@@ -0,0 +1,41 @@
...
+* password
+
+Once the user is created, you can add some information regarding his default language and his group.
+
+Group Management
+----------------
+
+In Open Orchestra, users can be members of groups.
+A group gives a user some rights and access for website.
+Each group has a name and is associated to only one website.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: Each group has a name, can be associated with only one website and contains a list of roles.

File: en/user_guide/user.rst [hunk 19]
```diff
@@ -0,0 +1,41 @@
...
+A role gives one type of authorization for a user.
+
+Role characteristics
+~~~~~~~~~~~~~~~~~~~~
+
+Roles contains:
+
+* name
+* description for each site languages:  to display in admin panel
+* original/destination status : optional parameters to sharp roles for specifics content types and define an edition line.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: I think that you should also put a link to the workflow documentation

File: en/user_guide/user.rst [hunk 20]
```diff
@@ -0,0 +1,48 @@
...
+
+Roles contains:
+
+* **name**
+* **description for each site languages**:  to display in admin panel
+* **original/destination status**: optional parameters to sharp roles for specifics content types and define an edition line.
+
+Some roles are already creating. By example:
+
+* **ROLE_ACCESS_USER**:  allows to manage `users
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: You forget to close the `

File: en/user_guide/user.rst [hunk 21]
```diff
@@ -0,0 +1,48 @@
...
+
+At the creation, some fields are required :
+
+* **first Name**
+* **last Name**
+* **e-mail**
+* **username**
+* **password**
+
+Once the user is created, you can add some information regarding his default language and his group.
```
Comment (veolys-nanne) [inline en/user_guide/user.rst]: some informations?

File: en/user_guide/user.rst [hunk 22]
```diff
@@ -0,0 +1,48 @@
...
+Role Management
+---------------
+
+A role gives one type of authorization for a user.
+
+Roles contains:
+
+* **name**
+* **description for each site languages**:  to display in admin panel
+* **original/destination status**: optional parameters to sharp roles for specifics content types and define an edition line.
```
Comment (veolys-nanne) [inline en/user_guide/user.rst]: for specific content types?

Comment (nicolasThal) [inline en/user_guide/user.rst]: can you also put a link to the workflow page?

File: en/user_guide/user.rst [hunk 23]
```diff
@@ -0,0 +1,48 @@
...
+
+Role Management
+---------------
+
+A role gives one type of authorization for a user.
+
+Roles contains:
+
+* **name**
+* **description for each site languages**:  to display in admin panel
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: **descriptions**: Name in every languages

File: en/user_guide/user.rst [hunk 24]
```diff
@@ -0,0 +1,48 @@
...
+* **original/destination status**: optional parameters to sharp roles for specifics content types and define an edition line.
+
+Some roles are already creating. By example:
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: `are already created`.

For exemple :

File: en/user_guide/user.rst [hunk 25]
```diff
@@ -0,0 +1,48 @@
...
+
+* **ROLE_ACCESS_USER**:  allows to manage users
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: do not put allows, I would propose `grant access to user management`

File: en/user_guide/user.rst [hunk 26]
```diff
@@ -0,0 +1,48 @@
...
+* **ROLE_ACCESS_REDIRECTION**: allows to manage redirections
+* **ROLE_ACCESS_ROLE**: allows to manage roles
+* **ROLE_ACCESS_SITE**: allows to manage websites
+* **ROLE_ACCESS_STATUS**: allows to manage status
+
+Group Management
+----------------
+
+Each group has a name, a list of roles and is associated to only one website.
+Users in a group have some rights and access for website define by group's roles.
```
Comment (ngilain) [inline en/user_guide/user.rst]: Maybe something like "Users in a group inherit rights defined in this group for the website associated"

File: en/user_guide/user.rst [hunk 27]
```diff
@@ -0,0 +1,53 @@
...
+
+Role Management
+---------------
+
+A role gives one type of authorization for a user.
+
+Roles contains:
+
+* **name**
+* **descriptions**: describe role in every languages to display in admin panel
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: `to display in admin panel` It sounds to me that something is missing.

You could suppress this part.

Or, the description is used when you have a form containing roles and the role description is displayed

File: en/user_guide/user.rst [hunk 28]
```diff
@@ -0,0 +1,53 @@
...
+
+Group Management
+----------------
+
+Each group has a name, a list of roles and is associated to only one website.
+Users in a group inherit rights defined in this group for the website associated.
+Using group is the simple way to give users some rights.
+
+
+.. _workflow:
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: I think that the workflow page is created ? 

I need @itkg-bfouche confirmation.

File: en/user_guide/user.rst [hunk 29]
```diff
@@ -0,0 +1,53 @@
...
+* **ROLE_ACCESS_ROLE**: grant access to role management
+* **ROLE_ACCESS_SITE**: grant access to websites management
+* **ROLE_ACCESS_STATUS**: grant access to status management
+
+Group Management
+----------------
+
+Each group has a name, a list of roles and is associated to only one website.
+Users in a group inherit rights defined in this group for the website associated.
+Using group is the simple way to give users some rights.
```
Comment (nicolasThal) [inline en/user_guide/user.rst]: `the simple way` sounds also wrong to me.

You could use `the recomanded way` or `the simpliest way`.

For the second guess, take care, because there is no other way to give roles to an user

File: en/user_guide/user.rst (hunk 28 above)
Comment (juchi) [inline en/user_guide/user.rst]: Indeed : /en/user_guide/workflow.rst

File: en/user_guide/user.rst [hunk 30]
```diff
@@ -0,0 +1,53 @@
...
+* **password**
+
+Once the user is created, you can add his default language and his groups.
+
+Role Management
+---------------
+
+A role gives one type of authorization for a user.
+
+Roles contains:
```
Comment (juchi) [inline en/user_guide/user.rst]: contain

File: en/user_guide/user.rst [hunk 31]
```diff
@@ -0,0 +1,53 @@
...
+
+* **name**
+* **descriptions**: describe role in every languages to display in admin panel
+* **original/destination status**: optional parameters to sharp roles for a content types and define a `workflow`_.
+
+Some roles are already created.
```
Comment (juchi) [inline en/user_guide/user.rst]: This is only true if the fixtures are loaded

File: en/user_guide/user.rst [hunk 32]
```diff
@@ -0,0 +1,53 @@
...
+* **ROLE_ACCESS_REDIRECTION**: grant access to redirection management
+* **ROLE_ACCESS_ROLE**: grant access to role management
+* **ROLE_ACCESS_SITE**: grant access to websites management
+* **ROLE_ACCESS_STATUS**: grant access to status management
+
+Group Management
+----------------
+
+Each group has a name, a list of roles and is associated to only one website.
+Users in a group inherit rights defined in this group for the website associated.
```
Comment (juchi) [inline en/user_guide/user.rst]: associated website

**Comment by nicolasThal** [comment]
:+1:

**Comment by itkg-rboessel** [comment]
@itkg-ngilain @itkg-bfouche @itkg-nanne

**Comment by itkg-bfouche** [comment]
:+1: with some pictures
